projects/media-viewer/src/lib/annotations/annotation-set/annotation-create/box-highlight-create.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, ElementRef, EventEmitter, Input, Output, ViewChild } from '@angular/core';
+import uuid from 'uuid';
+import { Subscription } from 'rxjs';
+import { BoxHighlightCreateService } from './box-highlight-create.service';
+
+@Component({
+  selector: 'mv-box-highlight-create',
+  templateUrl: './box-highlight-create.component.html',
+  styleUrls: ['./box-highlight-create.component.scss']
+})
+export class BoxHighlightCreateComponent {
+
+  @Input() pageHeight: number;
+  @Input() pageWidth: number;
+  @Input() rotate: number;
+  @Input() zoom: number;
+  @Input() container: { top: number, left: number };
+  @Output() highlightCreated = new EventEmitter();
+
+  @ViewChild('boxHighlight') highlight: ElementRef;
+
+  drawStartX = -1;
+  drawStartY = -1;
+  top: string;
+  left: string;
+  height: string;
+  width: string;
+  display: string;
+
+  private subscriptions: Subscription[] = [];
+
+  constructor(private readonly boxHighlightEvents: BoxHighlightCreateService) {}
+
+  ngOnInit(): void {
+    this.subscriptions = [
+      this.boxHighlightEvents.initHighlight.subscribe(event => this.init(event)),
+      this.boxHighlightEvents.updateHighlight.subscribe(event => this.update(event)),
+      this.boxHighlightEvents.createHighlight.subscribe(event => this.create())
+    ];
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(subscription => {
+      subscription.unsubscribe()
+    });
+  }
+
+  init(event) {
+    this.drawStartX = event.pageX - (window.pageXOffset + this.container.left);
+    this.drawStartY = event.pageY - (window.pageYOffset + this.container.top);
+
+    this.display = 'block';
+    this.height = '50px';
+    this.width = '50px';
+
+    switch (this.rotate) {
+      case 90:
+        this.top = this.pageHeight - this.drawStartX + 'px';
+        this.left = this.drawStartY + 'px';
+        break;
+      case 180:
+        this.top = this.pageHeight - this.drawStartY + 'px';
```

Comment:
i believe most of these require the width and height calculation to be involved.

Reply:
updated